### ISSUE TITLE
Support for config.subsplit-publish.json

### ISSUE THREAD

**Issue by alexander-schranz** [soyuka/pmu#1]
**Description**  

My root composer.json does not require to know anything about the repositories. It would be nice that "repositories" is not required and the supports the `config.subsplit-publish.json` file used by https://github.com/frankdejonge/use-subsplit-publish.

**Example**  

https://github.com/schranz-search/schranz-search/blob/0.3/config.subsplit-publish.json

**Comment by soyuka** [comment]
that's nice thanks will look into it, so according to how I do things, repositories would be added reading `[config.subsplit-publish.json](https://github.com/schranz-search/schranz-search/blob/0.3/config.subsplit-publish.json)` ?

**Comment by soyuka** [comment]
I'm not fond of this as I don't have the composer package names in that file. Instead, what about generating that `config.subsplit-publish.json` from the `composer.json` files? I think that all the information that are inside the `config.subsplit-publish.json` could be inside each of the `composer.json` (like the actual git repository).

**Comment by alexander-schranz** [comment]
> I'm not fond of this as I don't have the composer package names in that file.

You can get the `package name` the same way as for the repositories configuration of composer while you have there the `path` you have here `directory` configuration of the `config.subsplit-publish.json` and can read then `directory + /composer.json`. I don't think that it is required to generate here out of composer.json `config.subsplit-publish.json`.

**Comment by soyuka** [comment]
yes but right now I'm not reading the `composer.json` files myself as I want to leave that out to composer
